controllers/renovate_util.go

```diff
@@ -268,27 +308,127 @@ func (u ComponentDependenciesUpdater) GetUpdateTargetsGithubApp(ctx context.Cont
 			continue
 		}
 
+		customRenovateOptions, err := u.ReadCustomRenovateConfigMap(ctx, &component)
+		if err != nil {
+			log.Error(err, "failed to read custom renovate config map", "ComponentName", component.Name, "ComponentNamespace", component.Namespace)
+		}
+
+		// hardcoding the number in gitAuthor because mintmaker has it hardcoded as well, so that way mintmaker will recognize the same author
 		targetsToUpdate = append(targetsToUpdate, updateTarget{
-			ComponentName: component.Name,
-			GitProvider:   gitProvider,
-			Username:      fmt.Sprintf("%s[bot]", slug),
-			// hardcoding the number because mintmaker has it hardcoded as well, so that way mintmaker will recognize the same author
-			GitAuthor:               fmt.Sprintf("%s <126015336+%s[bot]@users.noreply.github.com>", slug, slug),
-			Token:                   githubAppInstallation.Token,
-			Endpoint:                git.BuildAPIEndpoint("github").APIEndpoint("github.com"),
-			Repositories:            repositories,
-			ImageRepositoryHost:     imageRepositoryHost,
-			ImageRepositoryUsername: imageRepositoryUsername,
-			ImageRepositoryPassword: imageRepositoryPassword,
+			ComponentName:                  component.Name,
+			ComponentCustomRenovateOptions: customRenovateOptions,
+			GitProvider:                    gitProvider,
+			Username:                       fmt.Sprintf("%s[bot]", slug),
+			GitAuthor:                      fmt.Sprintf("%s <126015336+%s[bot]@users.noreply.github.com>", slug, slug),
+			Token:                          githubAppInstallation.Token,
+			Endpoint:                       git.BuildAPIEndpoint("github").APIEndpoint("github.com"),
+			Repositories:                   repositories,
+			ImageRepositoryHost:            imageRepositoryHost,
+			ImageRepositoryUsername:        imageRepositoryUsername,
+			ImageRepositoryPassword:        imageRepositoryPassword,
 		})
 		log.Info("component to update for installations", "component", component.Name, "repositories", repositories)
 	}
 
 	return targetsToUpdate
 }
 
+// ReadCustomRenovateConfigMaps This method returns custom renovate options from config map
+func (u ComponentDependenciesUpdater) ReadCustomRenovateConfigMap(ctx context.Context, component *v1alpha1.Component) (CustomRenovateOptions, error) {
+	log := logger.FromContext(ctx)
+	customRenovateOptions := CustomRenovateOptions{}
+	customComponentRenovateConfigMap := readComponentCustomRenovateConfigMapAnnotation(component)
```

Comment:
`customComponentRenovateConfigMapName`.
nitpick: I'd put this line right before `if customComponentRenovateConfigMap != "" {`

Reply:
well it isn't just Name but actual configMap

Reply:
ok